### DISCUSSION TITLE
function in setInterval can not be update by watchtower, any advice pls

### DISCUSSION THREAD

**Discussion by ocylib** [containrrr/watchtower#982] [Q&A] [answered]
```
const a = async ()=>{
//do something here
}

const b= async()=>{
//do something else here
setInterval(async () => {
    const a = await a();
    console.log(a);
  }, 1000);
}

b()
```
When I updated my function a and then git push my code to github. Github actions will build and push a new image to docker hub. I have run watchtower and my own image in my server before I did this. 
I ran command : docker images, and found that the images is updated
I ran command: docker ps, and found that the container is updated
However, the log(I mean the result of function a ) was the same as before
The log would change until I ran command :docker restart myContainerId

I'd like to know why and how to solve this problem
Thank you

**Comment by piksel** [comment, marked as answer]
> However, the log(I mean the result of function a ) was the same as before

How did you read the logs? Someone mentioned something similar when using docker compose.

If you add `--debug` to the watchtower arguments you can see exactly when the old container is stopped and the new one is created.

**Reply by ocylib** [reply]
thanks for the reply. In fact, the actual code is more complicated than this, not just outputting a log. Except function a, I have a function named sendMsg(message) which makes a post request to a restful api, by doing this , send a message to my wechat app. Function sendMsg is outside of setInterval  like function a. In setInterval ,I pass the result of the A function as a parameter to the sendMsg function. when the container is updated I got same message (not updated) from my wechat app

**Reply by simonaronsson** [reply]
To me, this does not sound like a watchtower issue. If a new image was pulled, and the container was recreated, then it's running the new image, as far as I can imagine.

The only way to be completely sure however is by getting actual logs, as @piksel said.

**Reply by ocylib** [reply]
THANKS, IT SEEMS WORK FINE NOW.
